index.js

```diff
@@ -5,6 +5,32 @@ const sodium = require('sodium-native')
 const kObj = Symbol('object')
 const kCookieOptions = Symbol('cookie options')
 
+const proxyCache = new WeakMap()
```

Comment:
Why do you need a global WeakMap?

Reply:
It's shared between `sessionProxyHandler` and the function `createProxyHandler` which handles recursive object instances, so that calls to `get` are memoized.

Reply:
I have separated `WeakMap` into two different scopes as it might have caused confusion. I hope this approach is clearer.